src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -195,7 +204,166 @@ public void codeNoQuotes_multipleMatches() {
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: 2 spaces between keyword and quotes is not matched */
+    @Test
+    public void codeWithQuotes_2Spaces() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer code  \"bobat15\"";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: a line break between the keyword and code is acceptable */
+    @Test
+    public void codeWithQuotes_newLineBetween() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer code\n\"bobat15\"";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Arrays.asList("bobat15")));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: a line break inside the set of quotes will invalidate match */
+    @Test
+    public void codeWithQuotes_newLineInsideQuotes() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer code \"boba\nt15\"";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: keyword must come before the code, not after */
+    @Test
+    public void codeWithQuotes_lookbehind() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer \"bobat15\" code";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: keyword cannot be embedded inside another word */
+    @Test
+    public void codeWithQuotes_standaloneKeyword() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer decode \"bobat15\"";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: end match at first set of quotes encountered (lazy) */
+    @Test
+    public void codeWithQuotes_lazyMatching() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer code \"bobat\"15\"";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Arrays.asList("bobat")));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: also allow single quotes */
```

Comment:
Some of these function names are self explanatory.  I'm not sure if comments are needed.

Reply:
Removed unnecessary comments.